packages/react-examples/src/react/List/List.Basic.Example.tsx

```diff
@@ -5,80 +5,86 @@ import { TextField } from '@fluentui/react/lib/TextField';
-    {
-      minHeight: 54,
-      padding: 10,
-      boxSizing: 'border-box',
-      borderBottom: `1px solid ${semanticColors.bodyDivider}`,
-      display: 'flex',
-      selectors: {
-        '&:hover': { background: palette.neutralLight },
+const generateStyles = (theme: ITheme) => {
+  const { palette, semanticColors, fonts } = theme;
+  return mergeStyleSets({
+    itemCell: [
+      getFocusStyle(theme, { inset: -1 }),
+      {
+        minHeight: 54,
+        padding: 10,
+        boxSizing: 'border-box',
+        borderBottom: `1px solid ${semanticColors.bodyDivider}`,
+        display: 'flex',
+        selectors: {
+          '&:hover': { background: palette.neutralLight },
+        },
       },
+    ],
+    itemImage: {
+      flexShrink: 0,
     },
-  ],
-  itemImage: {
-    flexShrink: 0,
-  },
-  itemContent: {
-    marginLeft: 10,
-    overflow: 'hidden',
-    flexGrow: 1,
-  },
-  itemName: [
-    fonts.xLarge,
-    {
-      whiteSpace: 'nowrap',
+    itemContent: {
+      marginLeft: 10,
       overflow: 'hidden',
-      textOverflow: 'ellipsis',
+      flexGrow: 1,
     },
-  ],
-  itemIndex: {
-    fontSize: fonts.small.fontSize,
-    color: palette.neutralTertiary,
-    marginBottom: 10,
-  },
-  chevron: {
-    alignSelf: 'center',
-    marginLeft: 10,
-    color: palette.neutralTertiary,
-    fontSize: fonts.large.fontSize,
-    flexShrink: 0,
-  },
-});
-
-const onRenderCell = (item: IExampleItem, index: number | undefined): JSX.Element => {
-  return (
-    <div className={classNames.itemCell} data-is-focusable={true}>
-      <Image
-        className={classNames.itemImage}
-        src="https://res.cdn.office.net/files/fabric-cdn-prod_20230815.002/office-ui-fabric-react-assets/fluent-placeholder.svg"
-        width={50}
-        height={50}
-        imageFit={ImageFit.cover}
-      />
-      <div className={classNames.itemContent}>
-        <div className={classNames.itemName}>{item.name}</div>
-        <div className={classNames.itemIndex}>{`Item ${index}`}</div>
-        <div>{item.description}</div>
-      </div>
-      <Icon className={classNames.chevron} iconName={getRTL() ? 'ChevronLeft' : 'ChevronRight'} />
-    </div>
-  );
+    itemName: [
+      fonts.xLarge,
+      {
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      },
+    ],
+    itemIndex: {
+      fontSize: fonts.small.fontSize,
+      color: palette.neutralTertiary,
+      marginBottom: 10,
+    },
+    chevron: {
+      alignSelf: 'center',
+      marginLeft: 10,
+      color: palette.neutralTertiary,
+      fontSize: fonts.large.fontSize,
+      flexShrink: 0,
+    },
+  });
 };
 
 export const ListBasicExample: React.FunctionComponent = () => {
   const originalItems = useConst(() => createListItems(5000));
   const [items, setItems] = React.useState(originalItems);
+  const theme = useTheme();
+  const classNames = React.useMemo(() => generateStyles(theme), [theme]);
```

Comment:
I might be missing something but it looks like `classNames` is only used by `onRenderCell`?

You could move the computation of `classNames` into the `onRenderCell` `useCallback` and update the callback to depend on `theme` rather than `classNames`.

Reply:
Wouldn't that then call the class names computation a lot? Like, once for every cell?

Reply:
It would be possible to combine the two into a single `useMemo` call (no extra `useCallback`).
```ts
  const onRenderCell = React.useMemo(() => {
    const classNames = generateStyles(theme);
    return (item: IExampleItem, index: number | undefined): JSX.Element => {
      // ...
    };
  }, [theme]);
```

I don't think it gains enough to be worth the extra complexity for example code. IMO it's fine as-is.

Reply:
> Wouldn't that then call the class names computation a lot? Like, once for every cell?

It shouldn't be any different than now I think since the classnames computation would still only depend on the theme.